daemon/list.go

```diff
@@ -82,135 +156,152 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
+	}
 
-		img, err := daemon.Repositories().LookupImage(container.Config.Image)
-		if err != nil {
-			// If the image can no longer be found by its original reference,
-			// it makes sense to show the ID instead of a stale reference.
-			newC.Image = container.ImageID
-		} else if container.ImageID == img.ID {
-			newC.Image = container.Config.Image
-		} else {
-			newC.Image = container.ImageID
-		}
+	return includeContainer
+}
 
-		if len(container.Args) > 0 {
-			args := []string{}
-			for _, arg := range container.Args {
-				if strings.Contains(arg, " ") {
-					args = append(args, fmt.Sprintf("'%s'", arg))
-				} else {
-					args = append(args, arg)
-				}
-			}
-			argsAsString := strings.Join(args, " ")
+// transformContainer generates the container type expected by the docker ps command.
+func (daemon *Daemon) transformContainer(container *Container, ctx *listContext) (*types.Container, error) {
+	newC := &types.Container{
+		ID:    container.ID,
+		Names: ctx.names[container.ID],
+	}
 
-			newC.Command = fmt.Sprintf("%s %s", container.Path, argsAsString)
-		} else {
-			newC.Command = fmt.Sprintf("%s", container.Path)
-		}
-		newC.Created = container.Created.Unix()
-		newC.Status = container.State.String()
-		newC.HostConfig.NetworkMode = string(container.hostConfig.NetworkMode)
+	img, err := daemon.Repositories().LookupImage(container.Config.Image)
+	if err != nil {
+		// If the image can no longer be found by its original reference,
+		// it makes sense to show the ID instead of a stale reference.
+		newC.Image = container.ImageID
+	} else if container.ImageID == img.ID {
+		newC.Image = container.Config.Image
+	} else {
+		newC.Image = container.ImageID
+	}
 
-		newC.Ports = []types.Port{}
-		for port, bindings := range container.NetworkSettings.Ports {
-			p, err := nat.ParsePort(port.Port())
-			if err != nil {
-				return err
-			}
-			if len(bindings) == 0 {
-				newC.Ports = append(newC.Ports, types.Port{
-					PrivatePort: p,
-					Type:        port.Proto(),
-				})
-				continue
-			}
-			for _, binding := range bindings {
-				h, err := nat.ParsePort(binding.HostPort)
-				if err != nil {
-					return err
-				}
-				newC.Ports = append(newC.Ports, types.Port{
-					PrivatePort: p,
-					PublicPort:  h,
-					Type:        port.Proto(),
-					IP:          binding.HostIP,
-				})
+	if len(container.Args) > 0 {
+		args := []string{}
+		for _, arg := range container.Args {
+			if strings.Contains(arg, " ") {
+				args = append(args, fmt.Sprintf("'%s'", arg))
+			} else {
+				args = append(args, arg)
 			}
 		}
+		argsAsString := strings.Join(args, " ")
 
-		if config.Size {
-			sizeRw, sizeRootFs := container.GetSize()
-			newC.SizeRw = sizeRw
-			newC.SizeRootFs = sizeRootFs
-		}
-		newC.Labels = container.Config.Labels
-		containers = append(containers, newC)
-		return nil
+		newC.Command = fmt.Sprintf("%s %s", container.Path, argsAsString)
+	} else {
+		newC.Command = fmt.Sprintf("%s", container.Path)
```

Comment:
I agree, but I hope you noticed that you're commenting in code that has not being modified, I only moved it around :wink: